### ISSUE TITLE
[question] Customizing delimiters

### ISSUE THREAD

**Issue by vincent-lg** [prompt-toolkit/python-prompt-toolkit#1194]
Hi,

This might be a silly question, since I didn't find the answer in the documentation.  I really like this tool, great to build old-school console interfaces, and I'm always in for old-school console interfaces and was sorry to miss a tool for doing so in Python.  There's just one tiny feature that I miss, or perhaps, I couldn't find.

Boxes in this library use some delimiters to "look nice".  These are unicode characters.  They might look pretty nice on other consoles.  As it stands for me who uses a screen reader (yes, a console application is actually much more accessible with a screen reader), I would much prefer bare ASCII for delimiters.  Something with `|`, `-`, `\` and `/` which would not look nice for most users I guess.  Unicode symbols are read by the screen reader and displayed in Braille.  Not only do they "break" the console design, so it's hard to know where everything is, they also tend to slow down navigation pretty much.

So here's an example and here's what I "see":

```python
from prompt_toolkit.formatted_text import HTML
from prompt_toolkit.shortcuts import message_dialog
from prompt_toolkit.styles import Style

example_style = Style.from_dict({
    'dialog':             'bg:#88ff88',
    'dialog frame.label': 'bg:#ffffff #000000',
    'dialog.body':        'bg:#000000 #00ff00',
    'dialog shadow':      'bg:#00aa00',
})

message_dialog(
    title=HTML('<style bg="blue" fg="white">Styled</style> '
               '<style fg="ansired">dialog</style> window'),
    text='Do you want to continue?\nPress ENTER to quit.',
    style=example_style).run()
```

Straightforward example right from your documentation.  And now here's the screen I see:

                              ┌────────────────| Styled dialog window |────────────────┐
                              │                                                        │
                              │ Do you want to continue?                               │
                              │ Press ENTER to quit.                                   │
                              │                                                        │
                              │                      <    Ok    >                      │
                              │                                                        │
                              └────────────────────────────────────────────────────────┘

I guess you don't see the issue yet.  Let me add that `┌` is displayed by my screen reader like `U|J` (again, this is over-simplified and not 100% accurate), `─` somewhat appears like `U|w` and so on for every character used in the frame.  So it doesn't look like a frame to me, it looks like a lot of garbage with some understandable text a bit everywhere without logic.

Now, say, if we had an option for `─` to be a `-`, or `┌` to be a `/`, it would make things so easier to read.  I suggest adding an option when creating dialogs (and full screen windows) to disable unicode-delimiters and force delimiters to be ASCII only.

I hope I explained the situation in enough details and you see the problem I have.

Thanks again for this tool and your hard work,

**Comment by jonathanslenders** [comment]
Hi @vincent-lg,

Excuse me for the late reply.
I understand your problem and I'd very much like prompt_toolkit to be compatible with screen readers. That would be very nice.

If I understand correctly, it's not about limiting the whole application output to ASCII only (still allow Chinese characters to be printed for instance), but to not use box drawing characters and use ASCII symbols instead. Right?

Similarly to the ``PROMPT_TOOLKIT_COLOR_DEPTH`` environment variable, we could introduce a ``PROMPT_TOOLKIT_SCREEN_READER_SAFE`` variable (or something like that). I wonder actually whether this would be desired for other people to in some situations, in that case we should call it differently. Maybe ``PROMPT_TOOLKIT_BOX_DRAWING_STYLE``?

Does this apply to all screen readers? How do other frameworks handle this? Isn't there an option to configure the screen reader to not pronounce certain characters? I'm trying to understand the problem, so that we can come up with a good solution.

**Comment by vincent-lg** [comment]
Hi @jonathanslenders,

My turn to apologize for my late reply.  It is quite a busy time, as I'm sure we all experience.

I like your idea.  An environment variable wouldn't have been my first choice (Windows users, ugh) but they're clearly useful for the task and adding environment variables can be done in Python.  I think a broader name, as you suggested last, might be preferable, if only because others might find that useful whether they have a screen reader on or not.

So to answer your question: my need, I'm afraid, is a bit specific.  I use a Braille display in addition to my screen reader, for one, and that's not so common.  Screen readers can usually be set to handle unicode characters differently, so the frame shouldn't bother anyone with a speaking screen reader, either it will be ignored or the presentation won't matter at all.  To a Braille user, the problem is that unicode characters usually span more than one Braille cell.  For me, unicode characters that are not supported by my display are printed on the display as three distinct characters.  This creates a poor representation of a frame and upsets the nice alignment of data, making it hard to know where a thing is and what it is connected too.  That being said, I'm all for presentation and a Braille display does give a huge advantage in understanding Console interfaces, so I like to draw (yes, that's kind of odd).  So if the frame borders and other drawings were represented by ASCII characters like `|` or `-`, they would just appear as a single character on my Braille display and I would have a better picture of what the application does look like.

Why ASCII though?  This is a tricky question, because Braille does support Unicode representation.  But remember, a Braille cell only contains enough to represent, at the most, 256 characters.  Not exactly enough to represent Unicode.  So usually there's a compromise: ASCII retains mostly the same value in Braille, and additional unicode characters are represented, depending on whatever language you use for your Braille display.

In short, you are correct: I'm not suggesting to remove Chinese characters, I would simply encourage a way to turn the "unicode drawings" into "ASCII drawings", when they're meant to represent a frame for instance.

    \-------/
    | That  |
    |  is   |
    | cool, |
    | but   |
    | niños |
    | works |
    /-------\

Not sure I got the `/` and `\` right by the way, I keep confusing them.

Vincent
